src/client/Client.cc

```diff
@@ -5900,7 +5900,7 @@ int Client::may_setattr(Inode *in, struct ceph_statx *stx, int mask,
 	 *
 	 * Only allow unprivileged users to clear S_ISUID and S_ISUID.
 	 */
-	(m & ~(S_ISUID | S_ISGID)))
+	((m & ~(S_ISUID | S_ISGID)) || !(m & (S_ISUID | S_ISGID))))
```

Comment:
```
if (perms.uid() != 0 && perms.uid() != in->uid &&
    ((m & ~(S_ISUID | S_ISGID)) || !(m & (S_ISUID | S_ISGID))))
```
My reading of the above condition is that even if the file is owned by root, a non-privileged user can unset the `S_ISUID` and `S_ISGID` bits. But is that POSIXly correct ?
Please correct me if my understanding of the above condition is wrong.
Here's what I got to see during testing of this PR ...
```
mchangir:build$ echo hello > /mnt/mycephfs/greeting
mchangir:build$ ls -l /mnt/mycephfs/greeting
-rw-r--r--. 1 mchangir mchangir 6 Sep 20 12:30 /mnt/mycephfs/greeting
mchangir:build$ sudo chown root:root /mnt/mycephfs/greeting
mchangir:build$ ls -l /mnt/mycephfs/greeting
-rw-r--r--. 1 root root 6 Sep 20 12:30 /mnt/mycephfs/greeting
mchangir:build$ sudo chmod u+s /mnt/mycephfs/greeting
mchangir:build$ ls -l /mnt/mycephfs/greeting
-rwSr--r--. 1 root root 6 Sep 20 12:30 /mnt/mycephfs/greeting
mchangir:build$ chmod u-s /mnt/mycephfs/greeting
mchangir:build$ ls -l /mnt/mycephfs/greeting
-rw-r--r--. 1 root root 6 Sep 20 12:30 /mnt/mycephfs/greeting
```

Such a transition isn't allowed by btrfs on my `/home` partition.

Reply:
@mchangir I think you found a corner case bug. Thanks very much.

Then we must pass the `ATTR_KILL_SUID/ATTR_KILL_SGID` from kernel space via the libfuse to fix this issues, except that I couldn't find a better way to fix this issue.

Reply:
I could mostly find privileges dropping code in the kernel code during the `write` op.
Didn't go through it exhaustively.

You might want to add a small test case for this corner case.

Reply:
The xfstest-dev have already done that. So I think we can just use that instead of adding a new one.